src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClientManager.cs

```diff
@@ -57,19 +107,78 @@ internal ConfigurationClientManager(IList<ConfigurationClientWrapper> clients)
             _clients = clients;
         }
 
-        public IEnumerable<ConfigurationClient> GetAvailableClients(DateTimeOffset time)
+        public async IAsyncEnumerable<ConfigurationClient> GetAvailableClients([EnumeratorCancellation] CancellationToken cancellationToken)
         {
-            return _clients.Where(client => client.BackoffEndTime <= time).Select(c => c.Client).ToList();
-        }
+            DateTimeOffset now = DateTimeOffset.UtcNow;
+
+            // Principle of refreshing fallback clients:
+            // 1.Perform initial refresh attempt to query available clients on app start-up.
+            // 2.Refreshing when cached fallback clients have expired, FallbackClientRefreshExpireInterval is due.
+            // 3.At least wait MinimalClientRefreshInterval between two attempt to ensure not perform refreshing too often.
+
+            if (_replicaDiscoveryEnabled &&
+                now >= _lastFallbackClientRefresh + MinimalClientRefreshInterval && 
```

Comment:
the minimum interval must be respected between attempts. Right now `_lastFallbackClientRefresh` is being set after a success, meaning back to back failures can execute in short time spans. This is why I suggested to have one variable marking last attempt, and one marking last refresh.

Reply:
Yeah, I was thought that we could unify _lastFallbackClientRefresh and _lastFallbackClientRefreshAttempt. But I realized the short coming of it yesterday night after I committed the change, exactly it not good to the case you are saying. I will introduce _lastFallbackClientRefreshAttempt. and it will be updated before calling `RefreshFallabackClients`